drizzlepac/haputils/hapcut_utils.py

```diff
@@ -492,32 +485,89 @@ def mvm_combine(cutout_files, img_combiner=None, output_dir=".", log_level=logut
         The desired level of verbosity in the log statements displayed on the screen and written to the
         .log file. Default value is 20, or 'info'.
 
-    Note: The combiner only works with cutouts which have been generated as multiple output files,
-    rather than one large file with many cutout extensions.
-
     """
 
     # Make sure the cutout_files are really a list of MULTIPLE filenames
     if type(cutout_files) == str or type(cutout_files) == list and len(cutout_files) < 2:
         log.error("The 'mvm_combine' function requires a list of MULTIPLE cutout filenames where" \
-                  " the files were generated by 'make_the_cut' with the parameter single_output=False.")
+                  " the files were generated by 'make_the_cut'.")
+
+    # Sort the cutout filenames by detector (primary) and filter (secondary)
+    cutout_files.sort(key = lambda x: (x.split("_")[3], x.split("_")[4]))
 
     # Report the cutout files to be used for the combination process
     log.info("Input cutout files:")
     for cf in cutout_files:
-        log.info("{}".format(cf))
-
-    # Cutout filenames: hst_cutout_skycell-p0081-ra84d920799-decs69d851699[_uvis_f275w].fits
-    # Construct the combined filename based on the first file in the list as only RA and Dec are used
-    output_prefix = "hst_combined_skycells-"
-    filename = fits.getheader(cutout_files[0], ext=0)['FILENAME']
-    tokens = filename.split("_")[2].split("-")
-    skycell = tokens[1][1:5]
-    ra = tokens[2]
-    dec = tokens[3]
-    output_filename = os.path.join(output_dir, output_prefix + ra + "-" + dec + ".fits")
-
-    # Combine the SCI and then the WHT extensions in the specified files
-    log.info("Combining the SCI and then the WHT extensions of the input cutout files.")
-    combined_cutout = astrocut.CutoutsCombiner(cutout_files, img_combiner=img_combiner).combine(output_file=output_filename)
-    log.info("The combined output filename is {}.".format(output_filename))
+        log.info("File: {}".format(cf))
+
+    # Examples of input cutout filenames:
+    # hst_cutout_skycell-p0081x14y15-ra84d9207-decs69d8516_uvis_f275w.fits
+    # hst_cutout_skycell-p0081x14y15-ra84d9207-decs69d8516_wfc_f814w-jbp505jg.fits
+    # hst_cutout_skycell-p0081x14y15-ra84d9207-decs69d8516_ir_f128n_coarse.fits
+    # hst_cutout_skycell-p0081x14y15-ra84d9207-decs69d8516_ir_f128n_coarse-ibp505mf.fits
+
+    # Combined files will be generated for each detector/filter combination
+    # Walk the sorted input list and create a group list for each new detector/filter
+    temp_list = []
+    detector_filter_list = []
+    prev_detector = ""
+    prev_filter = ""
+    for cfile in cutout_files:
+        cf = cfile.replace(".fits", "")
+        detector = cf.split("_")[3]
+        filter = cf.split("_")[4].split("-")[0]
+        if detector != prev_detector:
```

Comment:
This logic lumps both filter and exposure level input files into 1 group, making it more difficult to combine just one or the other later, unless I am overlooking something...

Reply:
This has been addressed in commit 2343efc.